applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -887,132 +887,161 @@ static void init(void)
 	}
 }
 
-static bool event_handler(const struct event_header *eh)
+static bool handle_motion_event(const struct motion_event *event)
 {
-	if (is_motion_event(eh)) {
-		const struct motion_event *event = cast_motion_event(eh);
+	/* Do not accumulate mouse motion data */
+	state.last_dx = event->dx;
+	state.last_dy = event->dy;
 
-		/* Do not accumulate mouse motion data */
-		state.last_dx = event->dx;
-		state.last_dy = event->dy;
+	report_send(IN_REPORT_MOUSE, true, true);
 
-		report_send(IN_REPORT_MOUSE, true, true);
+	return false;
+}
+
+static bool handle_wheel_event(const struct wheel_event *event)
+{
+	state.wheel_acc += event->wheel;
+
+	report_send(IN_REPORT_MOUSE, true, true);
+
+	return false;
+}
+
+static bool handle_button_event(const struct button_event *event)
+{
+	/* Get usage ID and target report from HID Keymap */
+	struct hid_keymap *map = hid_keymap_get(event->key_id);
 
-		return false;
+	if (!map || !map->usage_id) {
+		LOG_WRN("No mapping, button ignored.");
```

Comment:
Remove dot

Reply:
it was there already :) this code is copy paste...

Reply:
Fixed